pkg/kubectl/resource_printer.go

```diff
@@ -353,103 +352,63 @@ func podHostString(host, ip string) string {
 // translateTimestamp returns the elapsed time since timestamp in
 // human-readable approximation.
 func translateTimestamp(timestamp util.Time) string {
-	return units.HumanDuration(time.Now().Sub(timestamp.Time))
-}
-
-// interpretContainerStatus interprets the container status and returns strings
-// associated with columns "STATUS", "CREATED", and "MESSAGE".
-// The meaning of MESSAGE varies based on the context of STATUS:
-//     STATUS: Waiting; MESSAGE: reason for waiting
-//     STATUS: Running; MESSAGE: reason for the last termination
-//     STATUS: Terminated; MESSAGE: reason for this termination
-func interpretContainerStatus(status *api.ContainerStatus) (string, string, string, error) {
-	// Helper function to compose a meaning message from terminate state.
-	getTermMsg := func(state *api.ContainerStateTerminated) string {
-		var message string
-		if state != nil {
-			message = fmt.Sprintf("exit code %d", state.ExitCode)
-			if state.Reason != "" {
-				message = fmt.Sprintf("%s, reason: %s", message, state.Reason)
-			}
-		}
-		return message
-	}
-
-	state := &status.State
-	if state.Waiting != nil {
-		return "Waiting", "", state.Waiting.Reason, nil
-	} else if state.Running != nil {
-		// Get the information of the last termination state. This is useful if
-		// a container is stuck in a crash loop.
-		message := getTermMsg(status.LastTerminationState.Terminated)
-		if message != "" {
-			message = "last termination: " + message
-		}
-		stateMsg := "Running"
-		if !status.Ready {
-			stateMsg = stateMsg + " *not ready*"
-		}
-		return stateMsg, translateTimestamp(state.Running.StartedAt), message, nil
-	} else if state.Terminated != nil {
-		return "Terminated", translateTimestamp(state.Terminated.StartedAt), getTermMsg(state.Terminated), nil
-	}
-	return "", "", "", fmt.Errorf("unknown container state %#v", *state)
+	shortHumanDuration := func(d time.Duration) string {
```

Comment:
That would also allow you to document its behavior.

I think the output could be a little simplified:

0s, 1s, 59s, 1m, 59m, 1h, 23h, 1d, 364d, 1y.

I don't think you need ~ or <. I also think you want to update at 23h, not 48h, to keep consistency.